e2e_playwright/st_chat_input_test.py

```diff
@@ -329,6 +329,30 @@ def test_uploads_and_deletes_multiple_files(
     expect(uploaded_file_names).to_have_text(files[1]["name"], use_inner_text=True)
 
 
+def test_single_file_upload_button_tooltip(
+    app: Page, assert_snapshot: ImageCompareFunction
+):
+    """Test that the single file upload button tooltip renders correctly."""
+    chat_input = app.get_by_test_id("stChatInput").nth(3)
+    chat_input.get_by_role("button").nth(0).hover()
+    wait_for_app_run(app, wait_delay=1500)
+
+    expect(app.get_by_text("Upload or drag and drop a file")).to_be_visible()
+    assert_snapshot(app, name="st_chat_input-single_file_button_tooltip")
```

Comment:
See the other comment; I think it's enough to just check that the tooltip with the correct text is visible. Doing fullscreen snapshots will probably create more trouble than value.